### PR TITLE
TweakPropertyGenerator Phase 6

### PR DESCRIPTION
**Pull request by albertodebortoli**
- Rename `configuration` to `tweakProvider` all over the place
- Rename `TweakPropertyGenerator` to `TweakAccessorGenerator`
- A lot more collateral renaming

### CODE REVIEW

File: Example/Tests/Core/LocalTweakProviderTests.swift [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import XCTest
+import JustTweak
+
+class LocalTweakProviderTests: XCTestCase {
+    
+    var tweakProvider: LocalTweakProvider!
+    
+    override func setUp() {
+        super.setUp()
+        tweakProvider = tweakProviderWithFileNamed("test_tweaks")!
```
Comment (agroGandhi) [inline Example/Tests/Core/LocalTweakProviderTests.swift:15]: should we try and do a similar name to the configuration file such as:
`LocalTweaks_test`

File: README.md [hunk 2]
```diff
@@ -219,16 +219,16 @@ The `TweakManager` provides the option to cache the tweak values in order to imp
 - post a `TweakConfigurationDidChangeNotification` notification
```
Comment (agroGandhi) [inline README.md:219]: Should we change the `TweakConfigurationDidChangeNotification`?

File: TweakAccessorGenerator/TweakAccessorGenerator/main.swift [hunk 3]
```diff
@@ -6,19 +6,19 @@
 import Foundation
 import ArgumentParser
 
-struct TweakPropertyGenerator: ParsableCommand {
+struct TweakAccessorGenerator: ParsableCommand {
     
     @Option(name: .shortAndLong, help: "The local configuration file path.")
```
Comment (agroGandhi) [inline TweakAccessorGenerator/TweakAccessorGenerator/main.swift:11]: maybe we should still include local here as we name the providers "LocalTweakProvider":
`localTweaksFilePath `

Also we need to update the help string

File: TweakAccessorGenerator/TweakAccessorGeneratorTests/Suites/TweakLoaderTests.swift [hunk 4]
```diff
@@ -21,10 +21,10 @@ class LocalConfigurationParserTests: XCTestCase {
     
     func test_loadConfiguration_success() throws {
         let bundle = Bundle(for: type(of: self))
-        let localConfigurationFilename = "ValidConfiguration"
-        let localConfigurationFilePath = bundle.path(forResource: localConfigurationFilename, ofType: "json")!
+        let tweaksFilename = "Tweaks"
```
Comment (agroGandhi) [inline TweakAccessorGenerator/TweakAccessorGeneratorTests/Suites/TweakLoaderTests.swift:24]: do we need `test_tweaks` and `test_tweaks_override` in the other tests? Filenames are lowercased but feels like we have a fair few files now

**Review by agroGandhi — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (albertodebortoli) [reply README.md:219]: Very well spotted! I had it on my list but forgot to tackle it!

File: Example/Tests/Core/LocalTweakProviderTests.swift (hunk 1 above)
Comment (albertodebortoli) [reply Example/Tests/Core/LocalTweakProviderTests.swift:15]: Yep!

File: TweakAccessorGenerator/TweakAccessorGeneratorTests/Suites/TweakLoaderTests.swift (hunk 4 above)
Comment (albertodebortoli) [reply TweakAccessorGenerator/TweakAccessorGeneratorTests/Suites/TweakLoaderTests.swift:24]: I've renamed them to `LocalTweaks_test.json` and `LocalTweaks_test_override.json`, still used in `TweakManager+PresentationTests.swift`, `LocalTweakProviderTests.swift`, `TweakManagerTests.swift` and `TweakViewControllerTests.swift`.
What we don't need anymore are these 2 files: `test_tweaks_invalid.json` and `test_tweaks_no_displayable_ungrouped.json`

**Review by agroGandhi — APPROVED** [review]

**Pull request merged**